taca_ngi_pipeline/deliver/deliver_grus.py

```diff
@@ -452,41 +429,86 @@ def _create_delivery_project(self, pi_id, sensitive):
             'api_opaque_data': '',
             'ngi_ready': False,
             'ngi_delivery_status': '',
-            'ngi_sensitive_data': sensitive
-        }
-
+            'ngi_sensitive_data': self.sensitive
+            }
+        if self.binfo_snic_id:
+            data['member_ids'] = [self.binfo_snic_id]
         response = requests.post(create_project_url, data=json.dumps(data), auth=(user, password))
         if response.status_code != 200:
             raise AssertionError("API returned status code {}. Response: {}. URL: {}".format(response.status_code, response.content, create_project_url))
         result = json.loads(response.content)
         return result
 
-    def _get_pi_id(self):
+    def _set_pi_details(self, given_pi_email=None):
+        """
+            Set PI email address and PI SNIC ID using PI email
+        """
+        self.pi_email, self.pi_snic_id = (None, None)
+        # try getting PI email
+        if given_pi_email:
+            logger.warning("PI email for project {} specified by user: {}".format(self.projectid, given_pi_email))
+            self.pi_email = given_pi_email
+        else:
+            try:
+                self.pi_email = self._get_order_detail()['project_pi_email']
+                logger.info("PI email for project {} found: {}".format(self.projectid, self.pi_email))
+            except Exception, e:
+                logger.error("Cannot fetch pi_email from StatusDB. Error says: {}".format(str(e)))
+                raise e
+        # try getting PI SNIC ID
+        try:
+            self.pi_snic_id = self._get_user_sinc_id(self.pi_email)
+            logger.info("SNIC PI-id for delivering of project {} is {}".format(self.projectid, self.pi_snic_id))
+        except Exception, e:
+            logger.error("Cannot fetch PI SNIC id using snic API. Error says: {}".format(str(e)))
+            raise e
+    
+    def _set_binfo_details(self):
```

Comment:
We were talking about having more contact people added automatically through the info the user makes in the order portal. If binfo_email was a list and we instead looped through it and tried to add all members of it then that would make it easier. 
Maybe rename `binfo_email` to something else in that case, like `extra_emails` or whatever.